src/osd/scheduler/OpScheduler.h

```diff
@@ -54,6 +54,10 @@ class OpScheduler {
   // Apply config changes to the scheduler (if any)
   virtual void update_configuration() = 0;
 
+  virtual double get_cost_per_io() const {
+    return 0.0;
```

Comment:
Does this actually get executed?  This should probably have a ceph_assert(0 == "impossible for wpq"); check to ensure that it isn't actually invoked.